tableslurp

```diff
@@ -313,12 +281,70 @@ def main():
         help='S3 bucket to download files from')
     ap.add_argument('origin', nargs=1,
         help='Path inside the bucket to the directory you want to download '
-        'files from')
+             'files from. '
+             'E.g if your bucket structure is <fqdn or name>/var/lib/cassandra/data/<ks>/<tbl>/<files>. \n'
+             'With --recursive, you should provide /var/lib/cassandra/data. \n'
+             'Otherwise, provide /var/lib/cassandra/data/<ks>/<tbl> if you want to restore just one table.')
     ap.add_argument('target', nargs=1,
         help='Path in the local FS where files should be downloaded to')
     args = ap.parse_args()
-    dh = DownloadHandler(args)
-    dh.run()
+
+    prefix='%s:%s' % (args.name, args.origin[0])
```

Comment:
The logic here is a little hard to follow... Maybe break it up into a few functions?

Reply:
I moved the fileset building into functions, and added some comments for `build_recursive_fileset`. Let me know if this is a Pythonic way of doing things. 

I also reverted the members that could be left as static to static to make the PR easier...but always feel free to point out bad python practices.

Reply:
Moved it into function `find_latest_listdirjson_for_every_table` which can be used by CLogs as well :)